sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ChangeFeedProcessorContext.java

```diff
@@ -25,4 +25,15 @@ public interface ChangeFeedProcessorContext {
      * */
     @Beta(value = Beta.SinceVersion.V4_51_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
     String getLeaseToken();
+
+    /**
+     * Get the diagnostics from the underlying feed response.
+     *
+     * @return The diagnostics object.
+     */
+    @Beta(value = Beta.SinceVersion.V4_64_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    default CosmosDiagnostics getDiagnostics()
+    {
+        return null;
```

Comment:
I wonder if throwing an exception `NotImplementedException` will make it more discoverable - `null` being returned can have impact later and callers may not diligently do null checks.

Reply:
+1

Reply:
fixed